tests/unittests/CoreText/CTFontManagerTests.mm

```diff
@@ -0,0 +1,110 @@
+    CFErrorRef error = nullptr;
+    EXPECT_TRUE(CTFontManagerRegisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> font{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, font);
+
+    woc::unique_cf<CGFontRef> cgfont{ CGFontCreateWithFontName(fontName.get()) };
+    EXPECT_NE(nullptr, cgfont);
+
+    StrongId<NSString> familyName = (NSString*)CTFontCopyFamilyName(font.get());
+    EXPECT_OBJCEQ(@"WinObjC", familyName);
+
+    EXPECT_TRUE(CTFontManagerUnregisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> failedFont{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, failedFont);
+
+    familyName = (NSString*)CTFontCopyFullName(failedFont.get());
+    EXPECT_OBJCEQ(@"Segoe UI", familyName);
+}
+
+TEST(CTFontManager, ShouldBeAbleToRegisterFontsForGraphicsFont) {
+    woc::unique_cf<CFStringRef> fontName{ CFSTR("WinObjC-Italic") };
+    NSURL* testFileURL = __GetURLFromPathRelativeToCurrentDirectory(@"/data/WinObjC-Italic.ttf");
+    woc::unique_cf<CGDataProviderRef> dataProvider{ CGDataProviderCreateWithURL((__bridge CFURLRef)testFileURL) };
+    woc::unique_cf<CGFontRef> graphicsFont{ CGFontCreateWithDataProvider(dataProvider.get()) };
+    EXPECT_NE(nullptr, graphicsFont);
+
+    CFErrorRef error = nullptr;
+    EXPECT_TRUE(CTFontManagerRegisterGraphicsFont(graphicsFont.get(), &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> font{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, font);
+
+    woc::unique_cf<CGFontRef> cgfont{ CGFontCreateWithFontName(fontName.get()) };
+    EXPECT_NE(nullptr, cgfont);
+
+    StrongId<NSString> familyName = (NSString*)CTFontCopyFullName(font.get());
+    EXPECT_OBJCEQ(@"WinObjC Italic", familyName);
+
+    EXPECT_TRUE(CTFontManagerUnregisterGraphicsFont(graphicsFont.get(), &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> failedFont{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, failedFont);
+
+    StrongId<NSString> failedFamilyName = (NSString*)CTFontCopyFullName(failedFont.get());
+    EXPECT_OBJCEQ(@"Segoe UI", failedFamilyName);
+}
+
+TEST(CTFontManager, ShouldFailToUnregisterNonregisteredFonts) {
+    NSURL* testFileURL = __GetURLFromPathRelativeToCurrentDirectory(@"/data/WinObjC.ttf");
+    CFErrorRef error = nullptr;
+    EXPECT_TRUE(CTFontManagerUnregisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(kCTFontManagerErrorNotRegistered, CFErrorGetCode(error));
+}
+
+TEST(CTFontManager, UIFontFamilyNamesShouldContainRegisteredFonts) {
+    NSArray* familyNames = [UIFont familyNames];
```

Comment:
Philosophical component isolation issue - UIKit is built on top of CT, so it's okay for UIKit tests to have CT, but not really vice-versa.
Additionally, for cross-compiling purposes, UIKit doesn't exist on OSX (only iOS), so we'd have to move this test out anyway if we were to make CoreText tests cross-compile.